File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& a) {
    int maxsum=0;
    int cursum =0;
        maxsum = a[0];

    for(int i=0;i<a.size(); i++)
        {
            cursum +=a[i];
            if(cursum>maxsum)
            {
              maxsum = cursum;
            }
            if(cursum<0) cursum = 0;    
         }
        return maxsum;
    }
};

File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum = -1e8
        currSum = 0
        for i in range(0, len(nums)):
            currSum = currSum + nums[i]
            if(currSum > maxSum):
                maxSum = currSum
            if(currSum < 0):
                currSum = 0
                         
        return maxSum